packages/wrangler/src/triggers/deploy.ts

```diff
@@ -240,6 +240,23 @@ export default async function triggersDeploy(
 		deployments.push(...updateConsumers);
 	}
 
+	if (config.workflows?.length) {
+		for (const workflow of config.workflows) {
+			deployments.push(
+				fetchResult(`/accounts/${accountId}/workflows/${workflow.name}`, {
+					method: "PUT",
+					body: JSON.stringify({
+						script_name: scriptName,
+						class_name: workflow.class_name,
+					}),
+					headers: {
+						"Content-Type": "application/json",
+					},
+				}).then(() => [`workflow: ${workflow.name}`])
+			);
+		}
+	}
```

Comment:
This change affects `wrangler triggers deploy` (which is used internally by `wrangler deploy`, `wrangler versions deploy`).

I was going to wrap these changes in an experimental flag (`–x-workflows`) but it seems redundant since there is already a if-block around it guarding on whether the user has created a `[[workflows]]` section in their wrangler.toml

Happy to add the flag (to all 3 commands) if anyone insists, but I've decided that it isn't necessary

Reply:
cc @lrapoport-cf

Reply:
I think we still need the flag. In fact, I think we should fail `wrangler.toml` validation if there is a `workflows` binding but no opt-in flag.

Reply:
Or else, the section in wrangler.toml should be called `experimental_workflows`...

Reply:
agree, let's still have the flag 👍 even though it's redundant, i think it still makes sense to keep so that users know they are using a new/separate pathway for workflows. 

after adding, if user has a workflow binding but doesn't specify the flag, will we error or warn? i'd lean towards being conservative and erroring, but happy to discuss

Reply:
https://github.com/cloudflare/workers-sdk/pull/7047#issuecomment-2429896208